### PR TITLE
Addon-actions: Add 'New Action' indicator

### PR DESCRIPTION
**Pull request by apapadopoulou**
Issue: #11124

## What I did

The issue I was aiming to solve was to add a 'New Action' indicator in the actions tab, so, whenever a new action is registered, a badge next to the actions title increments. The behaviour should be similar to the badge of the controls addon. The change was made in the actions register file, where the static title was replaced by a function that increments the badge value whenever an event that triggers an action occurs. If the story changes, the badge returns to the initial state (in our case, it disappears until a new action is made in the new story). 

There is, however, a problem in this approach. Every time an action-related event happens, the related function is subscribed to the render function too many times, and this could lead to performance issues and possibly memory leaks. Another approach would be to register the badge only once, by including it in a `React.useEffect()`. However, following this approach would lead to the badge being registered only once and then the new value is never displayed.

Maybe it would be best to get other opinions on the issue as well.

## How to test

I think an immediate validation would be to test it in the official example.

- Is this testable with Jest or Chromatic screenshots?
- Does this need a new example in the kitchen sink apps?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by shilman** [comment]
@apapadopoulou I made a change that fixes the re-rendering problem. I'm not sure if it's the most elegant way to do it, but it seems to work. @yannbf @tooppaaa WDYT?

File: addons/actions/src/register.tsx [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
 import ActionLogger from './containers/ActionLogger';
-import { ADDON_ID, PANEL_ID, PARAM_KEY } from './constants';
+import { ADDON_ID, EVENT_ID, PANEL_ID, PARAM_KEY } from './constants';
 
 addons.register(ADDON_ID, (api) => {
   addons.addPanel(PANEL_ID, {
-    title: 'Actions',
+    title() {
+      const [actionsCount, setActionsCount] = useState(0);
+      const onEvent = () => setActionsCount(actionsCount + 1);
```
Comment (tooppaaa) [inline addons/actions/src/register.tsx:11]: I'd probably do something like:
```suggestion
      const onEvent = () => setActionsCount(previous => previous + 1);
```

**Review by tooppaaa — COMMENTED** [review]
LGTM ! 🚀 
One minor comment.

**Review by shilman — APPROVED** [review]
Thank you @apapadopoulou @tooppaaa. Great addition! 💯

**Comment by yannbf** [comment]
Sorry for the late reply peeps! It's a great addition and finally fixes an issue that has been bugging people for almost a year. Super proud of you @apapadopoulou for making this happen!

**Comment by yannbf** [comment]
Hey @domyen I guess we can make the badge and include it in the next iteration of this addon!

**Pull request merged**